### PR TITLE
Implement a basic dynamic help mechanism

### LINKED ISSUE #155: Implement dynamic help system
Help should be enabled by a toggle button.
When enabled, all the "help-augmented" components should be highlighted on the UI, so that the user can ask help on a specific UI part. When an highlighted part is selected, the related help should be shown to the user.
We need to define how to help-augment any component and how to recognize them.

### PR DESCRIPTION
**Pull request by chrismayer**
This adds a dynamic help system to highlight 'help-augmented' elements
by a badge and show the corresponding help text by clicking the badge. The highlighting can be activated / deactivated by a button in the toolbar.

To show the mechanism the ZoomToMaxExtentButton is extended by such a "help-badge" 

Fixes #155

### CODE REVIEW

**Comment by mbarto** [comment]
Looks good. I was wondering if we can extend existing tools with help badge functionality using an Higher Order Component that wraps existing ones, instead of adding help functionality to every component one by one.
Something like:
<Help text="..."><ZoomToMaxExtent/></Help>

**Comment by chrismayer** [comment]
Thanks for you review @mbarto. I agree that a high-level wrapper would be good. I also discovered that it is not that good to extend every component and already tried some things to solve that, but no good approach so far.  I will have a deeper look on that.

**Comment by chrismayer** [comment]
I did some more investigation and added an approach for a high level component, which can be  wrapped around an existing element in order to show a help badge.
The positioning is a bit tricky because the ``HelpBadge`` is no more a direct child of the component itself. I solved it with a dynamic ID and appropriate CSS.
Any thoughts welcome.

File: web/client/components/Help/HelpWrapper.jsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        helpText: React.PropTypes.string,
+        helpEnabled: React.PropTypes.bool,
+        changeHelpText: React.PropTypes.func
+    },
+
+    render: function() {
+        return (
+            <div style={{position: "absolute"}}>
+                <HelpBadge
+                    id={"helpbadge-" + this.props.children.key}
```
Comment (mbarto) [inline web/client/components/Help/HelpWrapper.jsx]: We should add some check that one single child exist and has a key,

**Comment by mbarto** [comment]
Looks good to me, please proceed with other components.

**Comment by offtherailz** [comment]
Hi, It would be nice if the help text appears on mouse hover (on the ?) .
The interaction should be: the Help div appears on first hover event. The text changes on the next hover events, the help div disappear when the user disable the help. 
If would be cool if  the help div was draggable, maybe this can be a future enhancement, what do you think @mbarto ?

**Comment by chrismayer** [comment]
I added some more help badges to the app's components. To add them to the MapToolbar's buttons seems to be a bit more complicated since they are created dynamically and the ``<HelpWrapper>``-tag doesn't seem to be the right component, which is expected for in the MapToolbar class. Any hints welcome how to solve this in an elegant way.

The behaviour described by @offtherailz shouldn't be that hard to implement once a decision has been made.

**Comment by offtherailz** [comment]
I think we can do it passing some property helpMessage to each children and the other properties to the map toolbar directly. When the toolbar creates the buttons, it can wrap the component ( if the helpMessage property is present). This should solve the problem.

**Comment by mbarto** [comment]
Yes, I think it's a good idea. I also think  hover is the way to go.

**Comment by chrismayer** [comment]
I implemented the hover mechanism (f72225c) as described by @offtherailz:
>The interaction should be: the Help div appears on first hover event. The text changes on the next hover events, the help div disappear when the user disable the help.

**Comment by mbarto** [comment]
Hi Christian. Do you think you have still time to add the help badge also to the MapToolbar tools?
Can you also rebase your work on master, so that we don't have conflicts?
Thanks

**Comment by chrismayer** [comment]
Hi Mauro,
I already started, will provide it at the asap (no later than end of the day).

**Comment by chrismayer** [comment]
>Do you think you have still time to add the help badge also to the MapToolbar tools?
Can you also rebase your work on master, so that we don't have conflicts?

Done...

**Pull request merged**